x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/PhysicalPlanOptimizerTests.java

```diff
@@ -4023,6 +4024,25 @@ public void testRejectRemoteEnrichAfterCoordinatorEnrich() {
         );
     }
 
+    public void testNoStackOverflow() {
+        String query = randomBoolean() ? "row a = 1" : "row a = 1 | eval b = a";
+        for (int i = 0; i < 1900; i++) {
+            query += "::long::int";
+        }
```

Comment:
Should we check that no exception is thrown with the query this "short"?

Reply:
It could be a test by itself, but I'm not sure it never fails, also at that level. For sure most of the time it succeeds.